File: test-case/floating/matrix/matrix.go
package main

import (
    . "missingjs.com/soda/leetcode"
    "missingjs.com/soda/unittest"
    "missingjs.com/soda/util"
)

var logger = util.Logger
var _used ListNode

func matrixMultiply(a, b [][]float64) [][]float64 {
    var rows = len(a)
    var cols = len(b[0])
    var res = make([][]float64, rows)
    for i := range res {
        res[i] = make([]float64, cols)
    }

    for i := 0; i < rows; i++ {
        for j := 0; j < cols; j++ {
            var c = 0.0
            for k := 0; k < len(b); k++ {
                c += a[i][k] * b[k][j]
            }
            res[i][j] = c
        }
    }
    return res
}

func main() {
    // create tester by function
    work := unittest.CreateWork(matrixMultiply)
    // OR create by struct factory
    // work := unittest.CreateWorkForStruct(Constructor)
    // work.SetValidator(func(e,r)bool)
    // work.CompareSerial = true
    work.Run()
}

File: test-case/floating/matrix/matrix.ts
import {
    AvlTree,
    BinarySearchTree
} from '@datastructures-js/binary-search-tree';
import { TestWork, Utils, Validators } from 'soda/unittest';

// step [1]: implement solution function
/**
 * @param {number[][]} a
 * @param {number[][]} b
 * @return {number[][]}
 */
function matrixMultiply(a: number[][], b: number[][]): number[][] {
    let rows = a.length;
    let cols = b[0].length;
    let res = Array(rows).fill(0).map(() => Array(cols).fill(0));
    for (let i = 0; i < rows; ++i) {
        for (let j = 0; j < cols; ++j) {
            let c = 0;
            for (let k = 0; k < b.length; ++k) {
                c += a[i][k] * b[k][j];
            }
            res[i][j] = c;
        }
    }
    return res;
}

// step [2]: setup function/return/arguments
let taskFunc = matrixMultiply;
const work = TestWork.create<ReturnType<typeof taskFunc>>(taskFunc);
// work = TestWork.forStruct(STRUCT);
// work.validator = (x, y) => { ... };
// work.compareSerial = true;
console.log(work.run(Utils.fromStdin()));
